datafusion/common/src/scalar.rs

```diff
@@ -763,6 +763,42 @@ impl ScalarValue {
         }
     }
 
+    // ListArray compatible version of new_primitive
+    pub fn new_primitives<T: ArrowPrimitiveType>(
+        values: Vec<Option<T::Native>>,
+        d: &DataType,
+    ) -> Result<Self> {
+        if values.is_empty() {
+            return d.try_into();
+        }
+
+        // We need to convert it to ScalarValue::Primitive (Int64(0)) instead of ScalarValue::List (List([PrimitiveArray<Int64> [0]]))
+        if values.len() == 1 {
+            return Self::new_primitive::<T>(values[0], d);
```

Comment:
I think I would find this behavior very surprising -- that sometimes it returns a ScalarValue::List and sometimes a ScalarValue::Int8, etc -- I realize it is what your changes to sum need, but given it is so specialized, I think it would make more sense to check for a length of 1 at the callsite

Reply:
Note: I found `List([PrimitiveArray<Int64> [0]])` also represents the same as ScalarValue::Int64(0) in slt file, so I think we can skip this transformation